app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -121,22 +127,50 @@ class SecureMessageService @Inject()(
     }
   }
 
-  def addMessageToConversation(
+  def addCaseWorkerMessageToConversation(
+    client: String,
+    conversationId: String,
+    messagesRequest: CaseworkerMessageRequest)(
+    implicit ec: ExecutionContext,
+    hc: HeaderCarrier): Future[Either[SecureMessageError, Unit]] =
+    repo.conversationExists(client, conversationId).flatMap { exists =>
+      if (exists) {
+        messageContentCheck(messagesRequest.content).flatMap {
+          case Right(true) =>
+            getCaseworkerParticipant(client, conversationId, messagesRequest.sender.system.identifier).flatMap {
+              case Some(participant) =>
+                val message =
+                  Message(participant.id, new DateTime(), messagesRequest.content, isForwarded = Some(false))
+                createMessageAndSendEmail(client, conversationId, message, participant)
+              case _ => Future.successful(Left(NoCaseworkerIdFound("Caseworker ID not found")))
+            }
+          case Left(value) => Future.successful(Left(value))
+          case _           => Future.successful(Left(StoreError("Message not created", None)))
+        }
+      } else {
+        Future.successful(Left(ConversationIdNotFound("Conversation ID not known")))
+      }
+    }
+
+  def addCustomerMessageToConversation(
     client: String,
     conversationId: String,
     messagesRequest: CustomerMessageRequest,
-    enrolments: Enrolments)(implicit ec: ExecutionContext): Future[Unit] =
+    enrolments: Enrolments)(implicit ec: ExecutionContext): Future[Either[SecureMessageError, Unit]] =
     repo.conversationExists(client, conversationId).flatMap { exists =>
```

Comment:
i would move logic to repo.getConversation(clientId: String, conversationId): Future[Either[ConversationIdNotFound, Conversation]]
Then in the service you just add another line the the for comprehension :)